### PR TITLE
Use Qt 5.12.3 to work around macOS issues

### PR DESCRIPTION
**Pull request by JulianGro**
This PR works around the issues we are experiencing when mixing macOSXSDK10.12 with Qt 5.15.2.

It is a continuation of PR https://github.com/vircadia/vircadia/pull/1558

This has been CR'd up to 92907bd1221ecaab9007fe11e75c8700184b648c by @akamicah

### CODE REVIEW

File: tools/nitpick/src/TestCreator.cpp [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
 //  Copyright 2013 High Fidelity, Inc.
+//  Copyright 2022 Overte e.V.
 //
 //  Distributed under the Apache License, Version 2.0.
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 //
 #include "TestCreator.h"
 
 #include <assert.h>
+#include "../../../libraries/shared/src/QtCompatibility.h"
```
Comment (JulianGro) [inline tools/nitpick/src/TestCreator.cpp:14]: There must be a better way of handling this.

**Comment by AnimatedMannequin** [comment]
Here are some builds you may use for testing: 

**macOS-10.15, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.dmg

**self-hosted_debian-11_aarch64, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.tar.gz
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.tar.Z
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.sh

**ubuntu-18.04, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.tar.gz
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.tar.Z
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.sh

**windows-2019, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-51c743e.exe

**Comment by JulianGro** [comment]
DrFran reports this running on macOS Monterey

**Review by ksuprynowicz — APPROVED** [review]
I checked the changes and everything looks fine in my opinion.
I only have two questions:
Should `libraries/avatars/src/AvatarData.h` be empty?
Does `libraries/networking/src/udt/SendQueue.h` need `#include "QtCompatibility.h"`

**Comment by JulianGro** [comment]
> Should `libraries/avatars/src/AvatarData.h` be empty? 

What happened here is that this was changed and then changed back. I assume it only shows in the diff because some end-line character changed or something.

> Does `libraries/networking/src/udt/SendQueue.h` need `#include "QtCompatibility.h"`

Technically it only requires QtCompatibility.h on Qt versions lower than 5.15 because of "Q_DISABLE_COPY_MOVE"

**Comment by AnimatedMannequin** [comment]
Here are some builds you may use for testing: 

**macOS-10.15, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.dmg

**ubuntu-18.04, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.sh
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.tar.Z
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.tar.gz

**windows-2019, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.exe

**self-hosted_debian-11_aarch64, full**
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.sh
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.tar.Z
 - https://public.overte.org/build/overte/PR56/Overte-PR56-2f757aa.tar.gz

**Comment by ksuprynowicz** [comment]
It was just tested on M1 Mac Studio with Mac OS Monterey 12.3.1 and it works.
The executable is still called "Vircadia", even though it has Overte icon. Is there an easy way of changing it?

**Comment by JulianGro** [comment]
It's more hard coded than it should be.
Definitely something for another PR.

**Review by ksuprynowicz — APPROVED** [review]

**Comment by ksuprynowicz** [comment]
It crashes on MacOS High Sierra 10.13.6
[overte-log.txt](https://github.com/overte-org/overte/files/8543449/overte-log.txt)

**Comment by JulianGro** [comment]
The only clue I get from the log is that it stops logging very close to where the hifi.gl and hifi.gpu.gl should start logging.
Other than that, we would need a backtrace to get any more info.

I don't personally see a reason why this shouldn't run on High Sierra. Maybe something prebuilt like crashpad or webrtc is built for newer versions of macOS, but I don't see a warning indicating that.

**Pull request closed without merging**